modules/drivers/vertica/src/metabase/driver/vertica.clj

```diff
@@ -103,6 +110,22 @@
   [_ _ expr]
   (sql.qp/adjust-day-of-week :vertica (hsql/call :dayofweek_iso expr)))
 
+(defmethod sql.qp/->honeysql [:vertica :convert-timezone]
+  [driver [_ arg target-timezone source-timezone]]
+  (let [expr         (sql.qp/->honeysql driver arg)
+        timestamptz? (hx/is-of-type? expr "timestamptz")]
+    (when (and timestamptz? source-timezone)
+      (throw (ex-info (tru "`timestamptz` columns shouldn''t have a `source timezone`")
+                    {:type            qp.error-type/invalid-parameter
+                     :target-timezone target-timezone
+                     :source-timezone source-timezone})))
+    (let [source-timezone (or source-timezone (qp.timezone/results-timezone-id))]
+      (cond-> expr
+        (not timestamptz?)
+        (hx/->AtTimeZone source-timezone)
+        target-timezone
+        (hx/->AtTimeZone target-timezone)))))
```

Comment:
Add `with-type-info` here or equivalent.

Reply:
added in d9e419af69